apis/workloads/v1alpha1/helpers.go

```diff
@@ -214,11 +214,8 @@ matchRule:
 // HasAuthorisationRules defines whether a console template has authorisation
 // rules defined on it.
 func (ct *ConsoleTemplate) HasAuthorisationRules() bool {
-	if len(ct.Spec.AuthorisationRules) > 0 || ct.Spec.DefaultAuthorisationRule != nil {
-		return true
-	}
-
-	return false
+	return len(ct.Spec.AuthorisationRules) > 0 ||
+		(ct.Spec.DefaultAuthorisationRule != nil && ct.Spec.DefaultAuthorisationRule.AuthorisationsRequired > 0)
```

Comment:
I'm not convinced that we should be changing the behaviour of this function. It's also not related to the change in behaviour we are intending with this changeset.

In practice I think we should be moving to a place where there is always a default authorisation rule, even if this means defaulting the values to `0` to maintain backwards compatibility. At that point I think we should be removing this function in favour of more consistent logic to get the number of authorisations required for any given console.